server/src/main/java/org/elasticsearch/search/SearchHit.java

```diff
@@ -411,7 +456,12 @@ public String getSourceAsString() {
 
     @Override
     public Iterator<DocumentField> iterator() {
-        return fields.values().iterator();
+        // need to join the fields and metadata fields
+        Map<String, DocumentField> allFields = this.getFields();
+        if (allFields == null) {
```

Comment:
This can never be null, right? We always create a new map in getFields() currently.

Reply:
done